hudi-client/hudi-client-common/src/main/java/org/apache/hudi/client/BaseHoodieWriteClient.java

```diff
@@ -268,6 +268,7 @@ public boolean commitStats(String instantTime, HoodieData<WriteStatus> writeStat
       commitCallback.call(new HoodieWriteCommitCallbackMessage(
           instantTime, config.getTableName(), config.getBasePath(), stats, Option.of(commitActionType), extraMetadata));
     }
+    releaseResources(instantTime);
```

Comment:
this covers all writers correct? spark ds writer, and deltastreamer?

Reply:
yes, since SparkRDDWriteClient implements the method and is used across Spark DS and deltastreamer, we should be good.

Reply:
I guess we are missing async compaction flows. 
https://github.com/apache/hudi/blob/17ea14ab6d6a8ca7ecef2cfcdbc67b0c87f23987/hudi-client/hudi-spark-client/src/main/java/org/apache/hudi/client/HoodieSparkCompactor.java#L61
-> 
https://github.com/apache/hudi/blob/17ea14ab6d6a8ca7ecef2cfcdbc67b0c87f23987/hudi-client/hudi-client-common/src/main/java/org/apache/hudi/client/BaseHoodieWriteClient.java#L1045
-> 
https://github.com/apache/hudi/blob/17ea14ab6d6a8ca7ecef2cfcdbc67b0c87f23987/hudi-client/hudi-client-common/src/main/java/org/apache/hudi/client/BaseHoodieTableServiceClient.java#L322 

I don't think we are unpersisting rdds in this flow. 

Likely same for async clustering too.

Reply:
Fixed it for all table services, for both inline and async flows.

Reply:
good catch btw